### PR TITLE
feat: Adds the ability to delete the saved custom views

### LINKED ISSUE #3744: Add the ability to delete the folder

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes #3744 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screen recording** 

https://user-images.githubusercontent.com/64252451/149706327-fe6dd882-1e8a-42ae-864d-874a9a9d03ef.mov

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by fayazara — DISMISSED** [review]
LGTM

File: app/javascript/dashboard/components/ChatList.vue [hunk 1]
```diff
@@ -52,6 +64,14 @@
       @close="onCloseAddCustomViewsModal"
     />
 
+    <delete-custom-views
+      v-if="showDeleteCustomViewsModal"
+      :show-delete-popup.sync="showDeleteCustomViewsModal"
+      :close-delete-popup="onCloseDeleteCustomViewsModal"
```
Comment (nithindavid) [inline app/javascript/dashboard/components/ChatList.vue:70]: Make this an event `@close`

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/components/ChatList.vue (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/components/ChatList.vue]: Fixes by https://github.com/chatwoot/chatwoot/pull/3780/commits/d5e4f4609296da874efbb94dfa176554200fc980

File: app/javascript/dashboard/routes/dashboard/customviews/DeleteCustomViews.vue [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  computed: {
+    deleteMessage() {
+      return `${this.$t(
+        'FILTER.CUSTOM_VIEWS.DELETE.MODAL.CONFIRM.MESSAGE'
+      )} ${this.activeCustomView[0] && this.activeCustomView[0].name} ?`;
+    },
+    deleteConfirmText() {
+      return `${this.$t('FILTER.CUSTOM_VIEWS.DELETE.MODAL.CONFIRM.YES')} ${this
+        .activeCustomView[0] && this.activeCustomView[0].name}`;
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/customviews/DeleteCustomViews.vue:43]: Why are passing the prop `activeCustomView` as an array? It should be an object.

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/routes/dashboard/customviews/DeleteCustomViews.vue (hunk 2 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/customviews/DeleteCustomViews.vue]: FIxes by https://github.com/chatwoot/chatwoot/pull/3780/commits/d871b62e5a6aa6cba36f5f57616f85fa9a0a5adb

**Review by nithindavid — APPROVED** [review]

**Pull request merged**